person/src/homeworks/person/core/model/BasePerson.java

```diff
@@ -1,55 +1,114 @@
 package homeworks.person.core.model;
 
+import homeworks.person.core.exceptions.InvalidAgeException;
+import homeworks.person.core.exceptions.InvalidGenderException;
+import homeworks.person.core.exceptions.InvalidNameException;
 import homeworks.person.core.service.PersonAction;
+import homeworks.person.utils.Validate;
 
-public class BasePerson implements PersonAction {
+public abstract class BasePerson implements PersonAction {
     protected String name;
-    protected String designation;
+    protected String lastName;
+    protected String surname;
+    protected String nickname;
+    protected int age;
+    protected String email;
+    protected String  gender;
 
-    public BasePerson(String name, String designation)
+    private static final String nameRegexp = "^[A-Z][a-z]*$";
```

Comment:
@tatevikb please change name style to "UPPER_CASE"

Reply:
Done.